crates/telio-starcast/src/multicast_peer.rs

```diff
@@ -0,0 +1,131 @@
+use crate::{multicaster::Multicaster, receiver::Receiver};
+
+use boringtun::noise::Tunn;
+use std::{
+    net::{Ipv4Addr, Ipv6Addr, SocketAddr},
+    sync::Arc,
+};
+use telio_crypto::{PublicKey, SecretKey};
+use telio_sockets::SocketPool;
+use telio_task::io::Chan;
+use telio_utils::telio_log_error;
+use telio_wg::uapi::Peer;
+use telio_wg::DynamicWg;
+use tokio::{net::UdpSocket, sync::RwLock};
+use x25519_dalek::{PublicKey as PublicKeyDalek, StaticSecret};
+
+pub struct MulticasterPeer {
+    multicaster: RwLock<Multicaster>,
+    receiver: RwLock<Receiver>,
+}
+
+#[derive(Clone)]
+pub struct MulticasterIp {
+    pub ipv4: Ipv4Addr,
+    pub ipv6: Ipv6Addr,
+}
+
+impl MulticasterPeer {
+    pub async fn new(
+        telio_wg_port: u16,
+        public_key: &PublicKey,
+        wg: Arc<DynamicWg>,
```

Comment:
Should probably be Arc<dyn Wg>, and why is port needed if we can fetch it from wg?

Reply:
What should be Arc ? 
Port isn't needed. leftover.